src/Microsoft.AspNetCore.Http.Abstractions/HostString.cs

```diff
@@ -38,6 +64,88 @@ public bool HasValue
         }
 
         /// <summary>
+        /// Returns the value of the host part of the value. The port is removed if it was present.
+        /// IPv6 addresses will have brackets added if they are missing.
+        /// </summary>
+        /// <returns></returns>
+        public string Host
+        {
+            get
+            {
+                int index;
+                if (string.IsNullOrEmpty(_value))
+                {
+                    return string.Empty;
+                }
+                else if ((index = _value.IndexOf(']')) >= 0)
+                {
+                    // IPv6 in brackets [::1], maybe with port
+                    if ((index = _value.IndexOf(':', index + 1)) >= 0)
+                    {
+                        return _value.Substring(0, index);
+                    }
+
+                    return _value;
+                }
+                else if ((index = _value.IndexOf(':')) >= 0
+                    && index < _value.Length - 1
+                    && _value.IndexOf(':', index + 1) >= 0)
+                {
+                    // IPv6 without brackets ::1 is the only type of host with 2 or more colons
+                    return $"[{_value}]";
+                }
+                else if (index >= 0)
+                {
+                    // Has a port
+                    return _value.Substring(0, index);
+                }
+                else
+                {
+                    return _value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Returns the value of the port part of the host, or <value>0</value> if none is found.
+        /// </summary>
+        /// <returns></returns>
+        public int Port
```

Comment:
Yes, no magic values please. `var port = hostString.Port; if (port.HasValue)` is much better than `if (port != 0)`